internal/securitate/userTabler.go

```diff
@@ -0,0 +1,171 @@
+	_, err = tx.Exec(ctx, order, userName, tokenString)
+	if err != nil {
+		return fmt.Errorf("add TOKEN %w", err)
+	}
+	return tx.Commit(ctx)
+}
+
+func (dataBase *DBstruct) CheckUserPassword(ctx context.Context, userName, password string) error {
+	db := dataBase.DB
+	order := "SELECT (password = crypt($2, password)) AS password_match FROM " + "accounts" + " WHERE login= $1 ;"
+	row := db.QueryRow(ctx, order, userName, password) // password here - what was entered
+	var yes bool
+	err := row.Scan(&yes)
+	if err != nil {
+		return fmt.Errorf("QueryRow, error is %w", err)
+	}
+	if !yes {
+		return fmt.Errorf("password not match")
+	}
+	return nil
+}
+
+// nil - user exists
+func (dataBase *DBstruct) IfUserExists(ctx context.Context, userName string) error {
+	db := dataBase.DB
+	order := "SELECT 7 from " + "accounts" + " WHERE login= $1 ;"
+	row := db.QueryRow(ctx, order, userName) // password here - what was entered
+	var yes int
+	err := row.Scan(&yes)
+	if err != nil {
+		return fmt.Errorf(" QueryRow, error is %w", err)
+	}
+	if yes != 7 {
+		return fmt.Errorf("user %s does not exist", userName)
+	}
+	return nil
+}
+
+func (dataBase *DBstruct) ChangePassword(ctx context.Context, userName string, password string) error {
+	db := dataBase.DB
+	order := "UPDATE " + "accounts" + " SET password = crypt($2, gen_salt('md5')) WHERE login= $1 ;"
+	_, err := db.Exec(ctx, order, userName, password)
+	if err != nil {
+		return fmt.Errorf("change password error %w", err)
+	}
+	return nil
+}
+
+func (dataBase *DBstruct) UpdateToken(ctx context.Context, userName string, tokenString string) error {
+	db := dataBase.DB
+	order := "UPDATE tokens SET token = $2 WHERE userCode = (select usercode from accounts where login = $1) ;"
+	_, err := db.Exec(ctx, order, userName, tokenString)
+	if err != nil {
+		return fmt.Errorf("add TOKEN %w", err)
+	}
+	return nil
+}
+
+func (dataBase *DBstruct) GetToken(ctx context.Context, userName string, tokenString *string) error {
+	db := dataBase.DB
+	//				получить токен из токен-таблицы  где код пользователя равен коду юзера из юзер-таблицы с именем UserName
+	order := "SELECT token from " + "tokens" + " WHERE userCode = (select usercode from accounts where login = $1) ;"
+	row := db.QueryRow(ctx, order, userName)
+	var str string
+	err := row.Scan(&str)
+	if err != nil {
+		return fmt.Errorf("GT %w", err)
+	}
+	*tokenString = str
+	return nil
+}
+
+func (dataBase *DBstruct) UpLoadOrderByID(ctx context.Context, userID int64, orderNumber int64, orderStatus string) error {
+	db := dataBase.DB
+	order := "INSERT INTO orders(userCode, orderNumber, orderStatus, accrual) VALUES ($1, $2, $3, 0) ;" // accrual 0
+	_, err := db.Exec(ctx, order, userID, orderNumber, orderStatus)
+	if err != nil {
+		return fmt.Errorf("add ORDER %w", err)
+	}
+	return nil
+}
+
+func (dataBase *DBstruct) GetIDByOrder(ctx context.Context, orderNum int64, orderID *int64) error {
```

Comment:
```suggestion
func (dataBase *DBstruct) GetIDByOrder(ctx context.Context, orderNum int64) (int64, error) {
```
Нет никакого выигрыша в передаче возвращаемого параметра по ссылке в передаваемый параметр. Это даже звучит запутанно, выглядит точно так же. Вход на входе, выход на выходе - это нормальный Go-стиль.